packages/astro/src/assets/fonts/load.ts

```diff
@@ -117,27 +122,33 @@ export async function loadFonts({
 					typeof font.meta?.priority === 'number' ? font.meta.priority === 0 : true,
 				)
 				// Collect URLs
-				.map((font) => ({
-					...font,
-					src: font.src
-						// Limit src to 1 file (eg. if woff2 and woff are present, will only take woff2) to avoid
-						// downloading too many files
-						.slice(0, 1)
-						.map((source) =>
-							'name' in source
-								? source
-								: {
-										...source,
-										originalURL: source.url,
-										url: proxyURL({
-											value: source.url,
-											// We only use the url for hashing since the service returns urls with a hash already
-											hashString,
-											collect,
-										}),
-									},
-						),
-				}));
+				.map((font) => {
+					// The index keeps track of encountered URLs. We can't use the index on font.src.map
+					// below because it may contain sources without urls, which would prevent preloading completely
+					let index = 0;
+					return {
+						...font,
+						src: font.src.map((source) => {
+							if ('name' in source) {
+								return source;
+							}
+							const proxied = {
+								...source,
+								originalURL: source.url,
+								url: proxyURL({
+									value: source.url,
+									// We only use the url for hashing since the service returns urls with a hash already
+									hashString,
+									// We only collect the first URL to avoid preloading fallback sources (eg. we only
+									// preload woff2 if woff is available)
+									collect: (data) => collect(data, index === 0),
+								}),
+							};
+							index++;
+							return proxied;
+						}),
+					};
+				});
```

Comment:
nit: for readability, I would put this code before the return, and maybe use a for loop instead